sklearn/model_selection/tests/test_split.py

```diff
@@ -1038,36 +975,21 @@ def test_get_n_splits_for_repeated_stratified_kfold():
     assert expected_n_splits == rskf.get_n_splits()
 
 
-def test_repeated_stratified_kfold_determinstic_split():
-    X = [[1, 2], [3, 4], [5, 6], [7, 8], [9, 10]]
-    y = [1, 1, 1, 0, 0]
-    random_state = 1944695409
-    rskf = RepeatedStratifiedKFold(
-        n_splits=2,
-        n_repeats=2,
-        random_state=random_state)
-
-    # split should produce same and deterministic splits on
-    # each call
-    for _ in range(3):
-        splits = rskf.split(X, y)
-        train, test = next(splits)
-        assert_array_equal(train, [1, 4])
-        assert_array_equal(test, [0, 2, 3])
-
-        train, test = next(splits)
-        assert_array_equal(train, [0, 2, 3])
-        assert_array_equal(test, [1, 4])
-
-        train, test = next(splits)
-        assert_array_equal(train, [2, 3])
-        assert_array_equal(test, [0, 1, 4])
-
-        train, test = next(splits)
-        assert_array_equal(train, [0, 1, 4])
-        assert_array_equal(test, [2, 3])
+@pytest.mark.parametrize(
+    "RepeatedCV", [RepeatedKFold, RepeatedStratifiedKFold]
+)
+@pytest.mark.parametrize('random_state', [0, None, np.random.RandomState(0)])
+def test_repeated_folds_are_different(RepeatedCV, random_state):
+    # Make sure repeated folds are different at each repetition
 
-        assert_raises(StopIteration, next, splits)
+    X = np.arange(10).reshape(-1, 1)
+    y = [0] * 5 + [1] * 5
+    rkf = RepeatedCV(n_splits=2, n_repeats=2, random_state=random_state)
+    folds = list(rkf.split(X, y))
+    first_repetition = folds[:2]
+    second_repetition = folds[2:]
+    with pytest.raises(AssertionError):
+        np.testing.assert_equal(first_repetition, second_repetition)
```

Comment:
interesting!

Reply:
That's the desired behaviour right?? else it doesn't make much sense to use repeatedCV

Reply:
yeah sure. I was more surprised with having the assert the other way than the usual. We usually want the assert not to raise anything, and in this case my brain was like `assert np.any(first_repetition == second_repetition)`. Nothing to change, I was just confused for a moment when reading it.